t/is-and-extract-ipv6.pl

```diff
@@ -0,0 +1,531 @@
+	inet6 fdb6:1d86:d9bd:1:142c:8e9e:de48:843e prefixlen 64 autoconf secured
+	inet6 fdb6:1d86:d9bd:1:7447:cf67:edbd:cea4 prefixlen 64 autoconf temporary
+	inet6 fdb6:1d86:d9bd:1::c5b3 prefixlen 64 dynamic
+	inet6 2001:DB8:4341:0781:141d:66b9:2ba1:b67d prefixlen 64 autoconf secured
+	inet6 2001:DB8:4341:0781:64e1:b68f:e8af:5d6e prefixlen 64 autoconf temporary";
+# Sample output from MacOS when a manually configured static ULA is set.
+my $ifconfig_macos_static =
+"	inet6 fe80::1419:abd0:5943:8bbb%en0 prefixlen 64 secured scopeid 0xa
+	inet6 2001:DB8:4341:0781::101 prefixlen 64";
+my $ifconfig_macos_both = $ifconfig_macos . '\n' . $ifconfig_macos_static;
+
+# Sample output from RHEL: ifconfig <interface> | grep -w "inet6"
+my $ifconfig_rhel =
+"        inet6 2001:DB8:4341:0781::dc14  prefixlen 128  scopeid 0x0<global>
+        inet6 fe80::cd48:4a58:3b0f:4d30  prefixlen 64  scopeid 0x20<link>
+        inet6 2001:DB8:4341:0781:e720:3aec:a936:36d4  prefixlen 64  scopeid 0x0<global>
+        inet6 fdb6:1d86:d9bd:1:9c16:8cbf:ae33:f1cc  prefixlen 64  scopeid 0x0<global>
+        inet6 fdb6:1d86:d9bd:1::dc14  prefixlen 128  scopeid 0x0<global>";
+
+# Sample output from Ubuntu: ifconfig <interface> | grep -w "inet6"
+my $ifconfig_ubuntu =
+"        inet6 fdb6:1d86:d9bd:1:34a6:c329:c52e:8ba6  prefixlen 64  scopeid 0x0<global>
+        inet6 fdb6:1d86:d9bd:1:89b9:4b1c:186c:a0c7  prefixlen 64  scopeid 0x0<global>
+        inet6 fdb6:1d86:d9bd:1::8214  prefixlen 128  scopeid 0x0<global>
+        inet6 fdb6:1d86:d9bd:1:b417:fe35:166b:4816  prefixlen 64  scopeid 0x0<global>
+        inet6 fe80::5b31:fc63:d353:da68  prefixlen 64  scopeid 0x20<link>
+        inet6 2001:DB8:4341:0781::8214  prefixlen 128  scopeid 0x0<global>
+        inet6 2001:DB8:4341:0781:34a6:c329:c52e:8ba6  prefixlen 64  scopeid 0x0<global>
+        inet6 2001:DB8:4341:0781:89b9:4b1c:186c:a0c7  prefixlen 64  scopeid 0x0<global>
+        inet6 2001:DB8:4341:0781:f911:a224:7e69:d22  prefixlen 64  scopeid 0x0<global>";
+my $ifconfig_ubuntu_static =
+"        inet6 fe80::5b31:fc63:d353:da68  prefixlen 64  scopeid 0x20<link>
+        inet6 2001:DB8:4341:0781::100  prefixlen 128  scopeid 0x0<global>";
+my $ifconfig_ubuntu_both = $ifconfig_ubuntu . '\n' . $ifconfig_ubuntu_static;
+
+# Sample output from Busybox: ifconfig <interface> | grep -w "inet6"
+my $ifconfig_busybox =
+"          inet6 addr: fe80::4362:31ff:fe08:61b4/64 Scope:Link
+          inet6 addr: 2001:DB8:4341:0781:ed44:eb63:b070:212f/128 Scope:Global";
+
+my @if_samples = (
+    $ip_cmd_both,
+    $ifconfig_macos_both,
+    $ifconfig_rhel,
+    $ifconfig_ubuntu_both,
+    $ifconfig_busybox,
+);
+
+
+#######################################################################
+## Run through a bunch of IPv6 addresses
+foreach my $ip (@all_valid_ipv6) {
+    is(ddclient::is_ipv6($ip),1,"Testing valid 'is_ipv6($ip)'");
+}
+
+# Run through valid mixed (IPv4 within IPv6 address).  We do not accept
+# this format so expect valid check to fail.
```

Comment:
Why don't we? Short-term implementation convenience? If so, mark them as TODO.

Reply:
Permanent unless there is a need to set a mixed IPv6/IPv4 address to a DNS server.  I'm not even sure that it is possible.  Is it legal to set something like `2001:DB8::123.123.123.123` to a DNS server?  I've never seen that.  If it is possible then we can decide to support it, but for now it was deliberate choice not to.

Reply:
Okay, so I decided to try and set an AAAA record at cloudflare for a valid address inside my network `2601:xxxx:yyyy:e723::192.168.17.2` and see what happens.  Cloudflare accepted it and when I do a DNS request I get given the address `2601:xxxx:yyyy:e723::c0a8:1102` and pings to that appear to get routed correctly to my gateway... but of course don't get through because of my firewall.

Which means, I think we should support it.  But can we open it as an issue rather than hold up this PR please.

Reply:
> I think we should support it. But can we open it as an issue rather than hold up this PR please.

Sure. Just mark those as TODO:

```perl
for my $ip (@valid_mixed_ipv6) {
    TODO: {
        local $todo = "IPv6 addresses with embedded IPv4 addresses not yet supported";
        ok(ddclient::is_ipv6($ip), 1, "is_ipv6('$ip')");
    }
}
```

Reply:
Give me a day to look at it.  I think it might be fairly easy to add the support now.

Reply:
Mixed IPv6/IPv4 format now supported.  Ready for another review @rhansen